src/string.js

```diff
@@ -40,6 +40,19 @@ export function endsWith(string, suffix) {
 }
 
 /**
+ * Polyfill for String.prototype. startsWith.
+ * @param {string} string
+ * @param {string} prefix
+ * @return {boolean}
+ */
+export function startsWith(string, prefix) {
+  if (prefix.length > string.length) {
+    return false;
+  }
+  return string.indexOf(prefix) == 0;
```

Comment:
We can optimize this using a rather nifty trick:

```
return string.lastIndexOf(prefix, 0) === 0;
```

This prevents a full scan of the string, only checking index `0` (and possibly `1`, `2`, up to `prefix.length` if they match).